base/200-clusterrole.yaml

```diff
@@ -3,11 +3,24 @@ kind: ClusterRole
 apiVersion: rbac.authorization.k8s.io/v1
 metadata:
   name: tekton-dashboard-minimal
-  namespace: tekton-pipelines
   labels:
     app.kubernetes.io/component: dashboard
     app.kubernetes.io/instance: default
     app.kubernetes.io/part-of: tekton-dashboard
+aggregationRule:
+  clusterRoleSelectors:
+    - matchLabels:
+        rbac.authorization.k8s.io/aggregate-to-dashboard: "true"
+---
+kind: ClusterRole
```

Comment:
Can these rules not be kept as part of the `tekton-dashboard-minimal` ClusterRole above instead of splitting them out?

Reply:
I don't think so, aggregating overwrites the rules.

Reply:
https://kubernetes.io/docs/reference/access-authn-authz/rbac/#aggregated-clusterroles

> The aggregationRule defines a label selector that the controller uses to match other ClusterRole objects that should be combined into the rules field of this one.

I'll try this out locally but the wording above suggests it should merge them.

Reply:
I tried it and the result was the rules were overwritten.

I think the wording is confusing, the matching ClusterRoles rules are merged but the receiver rules are not.

I guess it makes sense as the object stored in etcd is the result of the aggregation, the original rules of the receiver are not preserved.

Reply:
Thinking about this some more it does make sense, the wording is just a little ambiguous. 'combined into the rules field' in this case means the ClusterRole objects matching the label are combined and the result then used to populate a rules field for this ClusterRole.

Reply:
hrmm now also wondering if the full-fat kustomize overlay falls foul of this